testing/bdd_test.ts

```diff
@@ -0,0 +1,1623 @@
+              assertEquals(it.ignore("example", { fn }), undefined);
+            }),
+        );
+
+        await t.step("all options", async () =>
+          await assertAllOptions((fn) => {
+            assertEquals(
+              it.ignore("example", {
+                fn,
+                ...baseOptions,
+              }),
+              undefined,
+            );
+          }));
+      });
+
+      await t.step(
+        "signature 3",
+        async () =>
+          await assertMinimumOptions((fn) => {
+            assertEquals(it.ignore("example", fn), undefined);
+          }),
+      );
+
+      await t.step(
+        "signature 4",
+        async () =>
+          await assertMinimumOptions((fn) => {
+            assertEquals(
+              it.ignore(function example(this: void, ...args) {
+                fn.apply(this, args);
+              }),
+              undefined,
+            );
+          }),
+      );
+
+      await t.step("signature 5", async (t) => {
+        await t.step(
+          "minimum options",
+          async () =>
+            await assertMinimumOptions((fn) => {
+              assertEquals(it.ignore("example", {}, fn), undefined);
+            }),
+        );
+
+        await t.step("all options", async () =>
+          await assertAllOptions((fn) => {
+            assertEquals(
+              it.ignore("example", {
+                ...baseOptions,
+              }, fn),
+              undefined,
+            );
+          }));
+      });
+
+      await t.step("signature 6", async (t) => {
+        await t.step(
+          "minimum options",
+          async () =>
+            await assertMinimumOptions((fn) => {
+              assertEquals(it.ignore({ name: "example" }, fn), undefined);
+            }),
+        );
+
+        await t.step("all options", async () =>
+          await assertAllOptions((fn) => {
+            assertEquals(
+              it.ignore({
+                name: "example",
+                ...baseOptions,
+              }, fn),
+              undefined,
+            );
+          }));
+      });
+
+      await t.step("signature 7", async (t) => {
+        await t.step(
+          "minimum options",
+          async () =>
+            await assertMinimumOptions((fn) => {
+              assertEquals(
+                it.ignore({}, function example(this: void, ...args) {
+                  fn.apply(this, args);
+                }),
+                undefined,
+              );
+            }),
+        );
+
+        await t.step("all options", async () =>
+          await assertAllOptions((fn) => {
+            assertEquals(
+              it.ignore({
+                ...baseOptions,
+              }, function example(this: void, ...args) {
+                fn.apply(this, args);
+              }),
+              undefined,
+            );
+          }));
+      });
+    });
+  });
+
+  await t.step("describe", async (t) => {
+    async function assertOptions(
```

Comment:
ditto: This test utility looks very complex. Please add doc comment which describes the purpose of this and what is checked with this.

Reply:
I can do that. These are just used internally. Basically every step in this block of tests have similar assertions regarding Deno.test and t.step getting called when you call describe and it, and this verifies Deno.test and t.step get called with the correct options.